src/OrchardCore/OrchardCore/Environment/Shell/RunningShellTable.cs

```diff
@@ -100,7 +100,7 @@ public ShellSettings Match(string host, string appRelativePath, bool fallbackToD
         private bool TryMatchInternal(string host, string appRelativePath, bool fallbackToDefault, out ShellSettings result)
         {
             // We can skip some checks for port matching if we know the host doesn't contain one
-            var hasPort = host.Contains(':');
+            var hasPort = host.IndexOf(':') != -1;
```

Comment:
LINQ trap, string doesn't have Contains.

Reply:
Okay, i got it.

Reply:
@lahma just saw [this in the corefx](https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/String.Searching.cs#L30). Do you think that `host.Contains(':')` is still a LINQ trap?

Reply:
Here I stand corrected, it's now part of netcoreapp3.0 and this PR targeted 2.2. So it's not trap when you know you target netstandard2.1 or netcoreapp3.0, but if you are library author then with netstandard2.0 or full framework targets, you still need to make sure where the call goes.

So it's a changing landscape, for Orchard it's now more idiomatic to just use the .Contains, thanks for pointing this out 👍

Reply:
No problem, we can let your change, i just wanted to be sure i understood things ;)

> but if you are library author then with netstandard2.0 or full framework targets, you still need to make sure where the call goes.

You're right!